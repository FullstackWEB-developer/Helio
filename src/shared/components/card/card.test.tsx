import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import {unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import React from 'react';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import utc from 'dayjs/plugin/utc';
import Card from '@components/card/card';

describe("Card tests", () => {
    let container: HTMLDivElement | null;
    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders card correctly", async () => {
        const {asFragment} = render(<Card>
            <div>Test</div>
        </Card>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders card with title correctly", async () => {
        const {asFragment} = render(<Card title='test'>
            <div>Test</div>
        </Card>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders card with extra correctly", async () => {
        const {asFragment} = render(<Card title='test' extra={<div>extra</div>}>
            <div>Test</div>
        </Card>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders card with radius correctly", async () => {
        const {asFragment} = render(<Card title='test' hasBorderRadius={true}>
            <div>Test</div>
        </Card>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders card with full height correctly", async () => {
        const {asFragment} = render(<Card title='test' hasFullHeight={true}>
            <div>Test</div>
        </Card>);
        expect(asFragment()).toMatchSnapshot();
    });

})
