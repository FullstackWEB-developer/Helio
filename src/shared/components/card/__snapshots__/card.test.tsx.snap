// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests renders card correctly 1`] = `
<DocumentFragment>
  <div
    class="card w-full"
  >
    <div
      class="w-full"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card tests renders card with extra correctly 1`] = `
<DocumentFragment>
  <div
    class="card w-full"
  >
    <div
      class="h-14 px-6 pt-5 h7 w-full flex justify-between"
    >
      <div>
        test
      </div>
      <div>
        <div>
          extra
        </div>
      </div>
    </div>
    <div
      class="w-full"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card tests renders card with full height correctly 1`] = `
<DocumentFragment>
  <div
    class="card w-full h-full"
  >
    <div
      class="h-14 px-6 pt-5 h7 w-full flex justify-between"
    >
      <div>
        test
      </div>
    </div>
    <div
      class="w-full"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card tests renders card with radius correctly 1`] = `
<DocumentFragment>
  <div
    class="card w-full rounded-xl"
  >
    <div
      class="h-14 px-6 pt-5 h7 w-full flex justify-between"
    >
      <div>
        test
      </div>
    </div>
    <div
      class="w-full"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card tests renders card with title correctly 1`] = `
<DocumentFragment>
  <div
    class="card w-full"
  >
    <div
      class="h-14 px-6 pt-5 h7 w-full flex justify-between"
    >
      <div>
        test
      </div>
    </div>
    <div
      class="w-full"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</DocumentFragment>
`;
